quickwit/quickwit-metastore/src/metastore/postgresql_metastore.rs

```diff
@@ -1284,9 +1294,110 @@ impl MetastoreService for PostgresqlMetastore {
 
     async fn open_shards(
         &mut self,
-        _request: OpenShardsRequest,
+        request: OpenShardsRequest,
     ) -> MetastoreResult<OpenShardsResponse> {
-        unimplemented!("`open_shards` is not implemented for PostgreSQL metastore")
+        let mut subresponses = Vec::with_capacity(request.subrequests.len());
+
+        let grouped_requests: HashMap<IndexUid, Vec<OpenShardsSubrequest>> = request
+            .subrequests
+            .into_iter()
+            .into_group_map_by(|subrequest| IndexUid::from(subrequest.index_uid.clone()));
+
+        for (index_uid, grouped_subrequests) in grouped_requests {
+            let index_metadata_request = IndexMetadataRequest {
+                index_id: None,
+                index_uid: Some(index_uid.to_string()),
+            };
+
+            let index_metadata: IndexMetadata = self
+                .index_metadata(index_metadata_request)
+                .await?
+                .deserialize_index_metadata()
+                .unwrap();
+
+            for subrequest in grouped_subrequests {
+                if !index_metadata.sources.contains_key(&subrequest.source_id) {
+                    return Err(MetastoreError::NotFound(EntityKind::Source {
+                        index_id: index_uid.index_id().to_string(),
+                        source_id: subrequest.source_id.to_string(),
+                    }));
+                }
+
+                let max_shard_opt: Option<PgShard> = sqlx::query_as::<_,PgShard>(
```

Comment:
This is done similar to the `FileBackedIndex` to ensure that we have a consistent implementation for all types.